src/Microsoft.DotNet.Build.Tasks/Targets/tests.targets

```diff
@@ -26,7 +26,12 @@
   <PropertyGroup Condition="'$(TestWithCore)' == 'true'">
     <TestRuntimeProjectJson Condition="'$(TestRuntimeProjectJson)' == ''">$(MSBuildThisFileDirectory)test-runtime\project.json</TestRuntimeProjectJson>
     <TestRuntimeProjectLockJson Condition="'$(TestRuntimeProjectLockJson)' == ''">$(MSBuildThisFileDirectory)test-runtime\project.lock.json</TestRuntimeProjectLockJson>
-    <TestHost>corerun.exe</TestHost>
+    <!-- Invoke with the correct casing of corerun per-OS.  There are some process tests
+         that check the invoked process name against known casing which fail under code coverage.
+         This is probably because the code coverage target invocation does not canonicalize the
+         process name. -->
+    <TestHost Condition="'$(OS)' == 'Windows_NT'">CoreRun.exe</TestHost>
+    <TestHost Condition="'$(OS)' != 'Windows_NT'">corerun.exe</TestHost>
```

Comment:
(Not that we're actually going to be running this as part of code coverage right now on Unix.)